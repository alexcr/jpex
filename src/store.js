import React from 'react';
import { useLocalStore } from 'mobx-react-lite';
import { createStore } from './createStore';

const storeContext = React.createContext(null);

export const StoreProvider = ({children}) => {
  const store = useLocalStore(createStore);
  return (
    <storeContext.Provider value={store}>
      {children}
    </storeContext.Provider>
  );
}

export const useStore = () => {
  const store = React.useContext(storeContext);
  if (!store) {
    throw new Error('useStore must be used within a StoreProvider.');
  }
  return store;
}
