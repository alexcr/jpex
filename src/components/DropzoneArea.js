import React from 'react';
import styled from 'styled-components';
import { useStore } from '../store';
import { useUpload } from '../upload';

const DropzoneContainer = styled.div`
  background-color: #f1ffec;
  padding-top: 15px;
  border: 1px solid #006616;
  border-top: 0px;
  font-size: 14px;
  font-family: 'Inconsolata', monospace;

  &:focus {
    outline: none;
  }

  &.dragging {
    background: url(icon.png) no-repeat center;
    background-position: calc(50% - 7px);
    background-size: 25%;
    background-color: #d8ffca;
  }
`;

export function DropzoneArea({children}) {
  const store = useStore();
  const {isDragAccept, getRootProps, getInputProps} = useUpload(store.setJSON);
  return (
    <DropzoneContainer
      className={isDragAccept ? 'dragging' : ''}
      {...getRootProps()}
    >
      <input {...getInputProps()} />
      {children}
    </DropzoneContainer>
  );
}
