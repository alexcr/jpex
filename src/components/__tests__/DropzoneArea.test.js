import React from 'react';
import { render } from 'enzyme';
import { DropzoneArea } from '../DropzoneArea';
import { StoreProvider } from '../../store';

describe('DropzoneArea', () => {
  it('should render a dropzone area correctly', () => {
    expect(
      render(
        <StoreProvider>
          <DropzoneArea>
            <div>Test Child</div>
          </DropzoneArea>
        </StoreProvider>
      )
    ).toMatchSnapshot();
  });
});
