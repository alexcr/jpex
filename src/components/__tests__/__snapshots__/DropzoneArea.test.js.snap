// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropzoneArea should render a dropzone area correctly 1`] = `
<div
  class="sc-bdnylx icGOsa"
  tabindex="0"
>
  <input
    accept="application/json"
    autocomplete="off"
    style="display:none"
    tabindex="-1"
    type="file"
  />
  <div>
    Test Child
  </div>
</div>
`;
